### PR TITLE
feat: send flow but animated

### PR DESCRIPTION
**Pull request by sandipndev**
## Description
This PR implements animations for Bitcoin Send flow.
- [x] Animations for all three forms: success (green check mark), eventual success (yellow clock) and failure (red cross).
- [x] Haptics at the time of send/failure
- [x] Minimum 2s for animation to complete even if the send is immediate (felt by the haptic) so that the UI feels reactive.

## Screenshots for types of payments
| Lighting Payment | Bitcoin Payment | Intraledger Payment |
|:--:|:--:|:--:|
| Simulator Screenshot - iPhone SE (3rd generation) - 2024-03-31 at 20 22 46 | Simulator Screenshot - iPhone SE (3rd generation) - 2024-03-31 at 20 23 33 | image |

## Recordings

Please wait a few seconds for the animation videos to load (~4 MB download).
Android animations performed slower in emulator (not an issue on device). So in these recordings you get to see how the animations will feel both slowly and fast)

| **"Send Success - Android" Animation** | **"Send Failure - Android" Animation** |
|:-----------:|:-----------:|
| Animation 1 | Animation 2 |

| **"Send Success - iOS" Animation** | **"Send Failure - iOS" Animation** |
|:-----------:|:-----------:|
| Animation 3 | Animation 4 |

---
All the animations were developed wholeheartedly by Liam. Thank you for your contribution, sir! ❤️

### CODE REVIEW

**Comment by nicolasburtey** [comment]
approving for now, even if I think the animation are somewhat over the top. I want to see how it feels when using the wallet. we can revert before next rollout if we're not on an agreement on what to do with this

File: app/assets/animations/Error Pulse Loop.json [hunk 1]
```diff
@@ -0,0 +1 @@
...:2,"bm":0,"ix":1,"mn":"ADBE Vector Group","hd":false}],"ip":261.000010630748,"op":264.00001075294,"st":235.000009571746,"bm":0},{"ddd":0,"ind":24,"ty":4,"nm":"Pulse Glitch 16","parent":1,"sr":1,"ks":{"o":{"a":0,"k":50,"ix":11},"r":{"a":0,"k":0,"ix":10},"p":{"a":0,"k":[50,51.094,0],"ix":2},"a":{"a":0,"k":[0.062,-297.938,0],"ix":1},"s":{"a":0,"k":[71,71,100],"ix":6}},"ao":0,"shapes":[{"ty":"gr","it":[{"d":1,"ty":"el","s":{"a":0,"k":[148.625,148.625],"ix":2},"p":{"a":0,"k":[0,0],"ix":3},"nm":"Ellipse Path 1","mn":"ADBE Vector Shape - Ellipse","hd":false},{"ty":"st","c":{"a":0,"k":[0.98431372549,0.337254901961,0.027450980392,1],"ix":3},"o":{"a":0,"k":100,"ix":4},"w":{"a":0,"k":4,"ix":5},"lc":1,"lj":1,"ml":4,"bm":0,"nm":"Stroke 1","mn":"ADBE Vector Graphic - Stroke","hd":false},{"ty":"tr","p":{"a":0,"k":[0.062,-297.938],"ix":2},"a":{"a":0,"k":[0,0],"ix":1},"s":{"a":0,"k":[122.534,122.534],"ix":3},"r":{"a":0,"k":0,"ix":6},"o":{"a":0,"k":100,"ix":7},"sk":{"a":0,"k":0,"ix":4},"sa":{"a":0,"k":0,"ix":5},"nm":"Transform"}],"nm":"Ellipse 1","np":3,"cix":2,"bm":0,"ix":1,"mn":"ADBE Vector Group","hd":false}],"ip":258.000010508555,"op":261.000010630748,"st":232.000009449554,"bm":0}],"markers":[]}
```
Comment (nicolasburtey) [inline app/assets/animations/Error Pulse Loop.json:1]: there shouldn't be space in filnemae

File: app/screens/send-bitcoin-screen/send-bitcoin-confirmation-screen.tsx [hunk 2]
```diff
@@ -137,72 +136,75 @@ const SendBitcoinConfirmationScreen: React.FC<Props> = ({ route }) => {
...
-        const indempotencyErrorPattern = /409: Conflict/i
-        if (indempotencyErrorPattern.test(err.message)) {
-          setPaymentError(LL.SendBitcoinConfirmationScreen.paymentAlreadyAttempted())
-          return
-        }
+        //   const indempotencyErrorPattern = /409: Conflict/i
+        //   if (indempotencyErrorPattern.test(err.message)) {
+        //     setPaymentError(LL.SendBitcoinConfirmationScreen.paymentAlreadyAttempted())
+        //     return
+        //   }
```
Comment (nicolasburtey) [inline app/screens/send-bitcoin-screen/send-bitcoin-confirmation-screen.tsx:202]: remove code commented

**Comment by nicolasburtey** [comment]
> approving for now, even if I think the animation are somewhat over the top. I want to see how it feels when using the wallet. we can revert before next rollout if we're not on an agreement on what to do with this

actually after skimming through the code, there is a lot of code commented, so have questions on that

File: app/screens/send-bitcoin-screen/send-bitcoin-confirmation-screen.tsx [hunk 3]
```diff
@@ -137,72 +136,75 @@ const SendBitcoinConfirmationScreen: React.FC<Props> = ({ route }) => {
...
+      //   })
+      //   return
+      // }
+
+      // setPaymentError(
+      //   errorsMessage || LL.SendBitcoinConfirmationScreen.somethingWentWrong(),
+      // )
+      // ReactNativeHapticFeedback.trigger("notificationError", {
+      //   ignoreAndroidSystemSettings: true,
+      // })
```
[the remark this answers is not in the document]
Comment (sandipndev) [reply app/screens/send-bitcoin-screen/send-bitcoin-confirmation-screen.tsx]: no, this logic has been moved to send-bitcoin-payment-screen

**Comment by sandipndev** [comment]
> approving for now, even if I think the animation are somewhat over the top. I want to see how it feels when using the wallet. we can revert before next rollout if we're not on an agreement on what to do with this

I've confirmed that there isn't any commented ghost code and have resolved your concerns on the PR. Based on your comment earlier, merging this to see and feel it in the app. We can revert before public if it feels over the top.

**Pull request merged**